query/src/common/evaluator/eval_context.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_datablocks::DataBlock;
+use common_datavalues::ColumnRef;
+use common_datavalues::DataField;
+use common_datavalues::DataTypeImpl;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::sql::IndexType;
+
+pub trait EvalContext {
+    type VectorID: PartialEq;
+
+    fn get_vector(&self, id: &Self::VectorID) -> Result<TypedVector>;
+
+    fn insert_vector(&mut self, id: Self::VectorID, vector: TypedVector) -> Result<()>;
+
+    fn tuple_count(&self) -> usize;
+}
+
+impl EvalContext for DataBlock {
+    type VectorID = IndexType;
+
+    fn get_vector(&self, id: &IndexType) -> Result<TypedVector> {
+        let name = id.to_string();
+        let column = self.try_column_by_name(name.as_str())?;
+        let field = self.schema().field_with_name(name.as_str())?;
```

Comment:
It seems that `column.data_type()` and `field.data_type()` are the same?

Reply:
`field.data_type()` is logical datatype.
`column.data_type()` is physical datatype.

Timestamp is logical datatype  as its physical datatype is int64.